src/plugins/intel_cpu/src/nodes/tensoriterator.h

```diff
@@ -65,33 +65,41 @@ class PortChecker {
 class DynamicBuffer {
 public:
     DynamicBuffer(const MemoryPtr &from_, const std::vector<MemoryPtr> &to_, const PortMap &map_rule_);
-    ~DynamicBuffer() = default;
 
     void execute(const dnnl::engine& eng, const int iter);
     void transfer(const Node* node);
 
+    void reset(int max_iter_count_);   // reset local
+
 private:
     void init(const dnnl::engine& eng);
 
     /* methods for resize and refill buffer */
-    dnnl::memory create_buffer(const dnnl::engine& eng);
-    void move_buffer(dnnl::memory new_buffer);
+    bool check_buffer();
+    MemoryPtr create_buffer(const dnnl::engine& eng);
+    void move_buffer(MemoryPtr& new_buffer);
```

Comment:
Either `const MemoryPtr& prim` or `MemoryPtr prim` depending on the ownership requirements.

Reply:
updated.